### PR TITLE
darshan-runtime: fix Job ID determination

### PR DESCRIPTION
**Pull request by Technohacker**
Regardless of the value assigned to the `scheduler` multi-variant, all checks would fail since they expected boolean variants instead

### CODE REVIEW

**Review by shanedsnyder — APPROVED** [review]

**Comment by shanedsnyder** [comment]
Looks good to me, thanks for the fix @Technohacker !

**Comment by Technohacker** [comment]
I recommend squash merging this PR because of that merge commit :sweat_smile:

File: var/spack/repos/builtin/packages/darshan-runtime/package.py [hunk 1]
```diff
@@ -100,14 +100,15 @@ def configure_args(self):
         spec = self.spec
         extra_args = []
 
+        scheduler = spec.variants["scheduler"].value
         job_id = "NONE"
-        if spec.satisfies("+slurm"):
+        if scheduler == "slurm":
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/darshan-runtime/package.py]:
```suggestion
        if spec.satisfies("scheduler=slurm"):
```
and similarly below

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/darshan-runtime/package.py (hunk 1 above)
Comment (Technohacker) [reply var/spack/repos/builtin/packages/darshan-runtime/package.py]: Fixed, thanks!

**Review by alalazo — APPROVED** [review]

**Pull request merged**